### ISSUE TITLE
Can't use Unix Signals inside a worker, the worker is automatically killed

### ISSUE THREAD

**Issue by LilianCr** [ray-project/ray#15661]
### What is the problem?
I am not able to use Unix signals inside a Ray worker.
Each time ANY Unix signal is raised inside a worker, the worker is automatically killed.
Thanks in advance for your help :slightly_smiling_face: 

### Reproduction (REQUIRED)
```python3
import ray
import signal

ray.init()

@ray.remote
def my_simple_worker():
     signal.raise_signal(50)

my_simple_worker.remote()
```

**Output :**
```
>>> python3 test.py
2021-05-06 17:27:29,957	INFO services.py:1267 -- View the Ray dashboard at http://127.0.0.1:8265
2021-05-06 17:27:36,254	WARNING worker.py:1115 -- A worker died or was killed while executing task a67dc375e60ddd1affffffffffffffffffffffff01000000.
```

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [X] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by rkooo567** [comment]
Can you briefly state what's the use case of raising signals (without having a separate handler)?

**Comment by LilianCr** [comment]
> Can you briefly state what's the use case of raising signals (without having a separate handler)?

The goal is to have very fast IPC communication with very low latency (less than 0.1 ms) using UNIX signals to notify other processes that new datas are available and using mutltiprocessing.share_memory to host datas.

I did some tests today and if I use `os.kill(pid, signal)` to send a signal from a worker A to a worker B, the worker B is automatically killed.
When `signal.raise_signal(signal)` is called the signal is sent to the current process so if a worker A send a signal with `signal.raise_signal(signal)` the worker A is automatically killed because he sent a signal to himself.

I think it could be a good behavior to only kill workers when signals like SIGKILL are received.

Would it be possible ?
Is there any work around / hack I could use temporarily ?

**Comment by LilianCr** [comment]
The problem was on my side, I'm going to close the issue.
Thanks for your time @rkooo567 :slightly_smiling_face: 

Here is my how I solved the problem (if someone was in the same case a day) :
Firstly I executed `signal.getsignal(pid)` and the default handler `signal.SIG_DFL` was returned.
This means 2 things :
- My own handler was not working,
- The problem doesn't come from Ray or in other words, Ray has not set its own handler.

Then I just tried to use `signal.signal(pid, handler)` to catch Unix signals and the worker was no more killed,  `signal.getsignal(pid)` returned my own handler.
I'm looking for an async method to receive signals (that was why I created my own handler) and at the moment I found [Asyncio Signals Handler](https://pymotw.com/3/asyncio/unix_signals.html) and seems to works fine with Ray too.
